src/classes/OPP_OpportunityNaming_BATCH.cls

```diff
@@ -34,22 +34,39 @@
 * @group-content ../../ApexDocContent/Opportunity.htm
 * @description Batch class creates names all Opportunities per the naming spec.
  */
-public class OPP_OpportunityNaming_BATCH implements Database.Batchable<sObject>, Schedulable {
-    
+public class OPP_OpportunityNaming_BATCH implements Database.Batchable<sObject>, Schedulable, Database.Stateful {
     /** @description The query for the batch process to run on.*/
-    String query;
-    
+    private String query;
+
+    @TestVisible
+    private static Integer defaultQueryLimit = 10000000;
```

Comment:
What would you think of making this a constant?
`private static final Integer DEFAULT_QUERY_LIMIT = 10000000;`

Reply:
I didn't make this a constant only because I wanted to be able to change it during tests so that I can test the chunking in `OPP_OpportunityNaming_BATCH.cls.shouldRenameOpportunitiesInChunks`

Reply:
Very good point.